cmd/api/src/services/ingest/job_integration_test.go

```diff
@@ -0,0 +1,37 @@
+package ingest
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/specterops/bloodhound/src/model"
+)
+
+func TestIntegration_SaveIngestFile_JSON(t *testing.T) {
+	// TODO: Make this
+	body := io.NopCloser(bytes.NewBufferString(`{"meta":{"methods":46067,"type":"computers","count":0,"vers ion":6},"data":[]}`))
+
+	// TODO: Make this take it a FS?
+	location, fileType, err := SaveIngestFile("/tmp", "application/json", body)
+	if err != nil {
+		t.Fatalf("SaveIngestFile failed: %v", err)
+	}
+
+	if fileType != model.FileTypeJson {
+		t.Errorf("expected fileType %v, got %v", model.FileTypeJson, fileType)
+	}
+
+	info, statErr := os.Stat(location)
+	if statErr != nil {
+		t.Fatalf("output file not found: %v", statErr)
+	}
+
+	if info.Size() == 0 {
+		t.Error("file saved is empty")
+	}
```

Comment:
We have access to testify for `require` and `assert` assertions, it makes the test code a bit more concise and easier to reason about. I'd recommend using `require.NoError` for your error conditions that should never fail because they'd cause the remaining assertions to autofail, and appropriate assertions like `assert.Equal` to check size. See https://pkg.go.dev/github.com/stretchr/testify/assert#pkg-functions for more docs on what assertions exist. The more specific the better, since it acts as documentation for what specifically you're asserting.